### PR TITLE
Variable change

### LINKED ISSUE #79: Deprecate the tower_auth_secret in favor of a new connection_secret
We should rename the `tower_auth_secret` to `connection_secret`.  In order to not break existing deployments though, tower_auth_secret should be deprecated, but logic to conditionally use either should be added.  

It should preferentially use `connection_secret` if provided.

### PR DESCRIPTION
**Pull request by cooktheryan**
Closes #79

### CODE REVIEW

**Comment by cooktheryan** [comment]
@rooftopcellist so with this we can still use the old secret parameter for jobs and jobtemplates for the new crds we will only accept the connection_secret flag.

**Comment by rooftopcellist** [comment]
@cooktheryan what do you think about making this bit a playbook in a new `roles/common` which gets include for the roles that need it?
* https://github.com/ansible/awx-resource-operator/pull/112/files#diff-1150474f1bb6d092f91803bc24bbce95611ac39c7ca22079154de597fb7a8807L2-R34

File: roles/workflowtemplate/tasks/main.yml [hunk 1]
```diff
@@ -1,10 +1,37 @@
...
+
+- name: Set fact based on secret results
+  set_fact:
+     tower_config_secret: "{{ tower_connection_secret }}"
+  when: tower_connection_secret is not skipped and tower_connection_deprecated_secret is skipped
+
+- name: Set fact based on secret results
+  set_fact:
+    tower_config_secret: "{{ tower_connection_deprecated_secret }}"
+  when: tower_connection_deprecated_secret is not skipped and tower_connection_secret is skipped
```
Comment (rooftopcellist) [inline roles/workflowtemplate/tasks/main.yml:28]: @cooktheryan These tasks only seem to affect the following task that checks if the secret-to-be-used exists.  

Don't we similarly need logic to set `connection_secret` to be the name of the secret to be used (`tower_auth_secret` or `connection_secret`) before job_definition.yml.j2 is applied?

Comment (cooktheryan) [reply roles/workflowtemplate/tasks/main.yml:28]: good catch...will patch up and add an extra test case as well 👍

**Pull request closed without merging**